grant/grant.go

```diff
@@ -495,6 +503,43 @@ func (req *RedeemGrantsRequest) VerifyAndConsume(ctx context.Context) (*wallet.T
 	return &redeemTxInfo, nil
 }
 
+// GetRedeemed returns the redemption status for given grants under a provider id
+func (req *RedeemGrantsRequest) GetRedeemed(ctx context.Context) ([]Grant, error) {
+	grants, err := req.DecodeGrants()
+	if err != nil {
+		return nil, err
+	}
+
+	grantCount := len(grants)
+	results := make([]Grant, 0, grantCount)
+
+	for _, grant := range grants {
+		promotionIDString := grant.PromotionID.String()
+		grantIDString := grant.GrantID.String()
+		storeKey := grantKeyClaim(promotionIDString)
```

Comment:
this is kind of an awkward abstraction, can we instead make a `GetPromotionGrantsDatastore` helper function (and `GetPromotionWalletsDatastore` to match I guess)

Reply:
`func GetPromotionGrantsDatastore(ctx context.Context, promotionID uuid.UUID) (SetLikeDatastore, error)`